### PR TITLE
Remove deprecated testscases for pam

### PR DESCRIPTION
**Pull request by punkioudi**
As pam_tally2 and pam_cracklib modules are deprecated for >=1.5.0 pam releases, the corresponding tests fail in the testsuite, so they have to be removed

- Related ticket: https://progress.opensuse.org/issues/99039
- Needles: N/A
- Verification run: [Tumbleweed](https://openqa.opensuse.org/tests/2016700#step/pam/41)
- [sles15sp3](https://openqa.suse.de/tests/7613725#step/pam/52) | [sles15sp2
](https://openqa.suse.de/tests/7613728#step/pam/52) | [sles15sp1](https://openqa.suse.de/tests/7613787#step/pam/52) | [sles15](https://openqa.suse.de/tests/7613791#step/pam/52) | [sles12sp5](https://openqa.suse.de/tests/7613792#step/pam/52) | [sles12sp4](https://openqa.suse.de/tests/7613793#step/pam/52) | [sles12sp3](https://openqa.suse.de/tests/7613794#step/pam/52) | [sles12sp2](https://openqa.suse.de/tests/7613795#step/pam/52)

### CODE REVIEW

File: tests/console/pam.pm
Comment (DimStar77) [inline tests/console/pam.pm:40]: group(wheel) should probably remain - as we still have the test:

```
@test "modify group access" {                                         # case 17
    echo "*;*;$USER_NOR;Al0000-2400;wheel" >> /etc/security/group.conf
```

**Comment by DimStar77** [comment]
Looks almost identical to https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13307 :)

File: tests/console/pam.pm
Comment (punkioudi) [reply tests/console/pam.pm:40]: > Looks almost identical to #13307 :)

Oh indeed! Hm, what I see in that in sle, the tests don't fail yet, as  the pam version is older than the one that these 2 modules are being deprecated (eg for sle15sp3, I see `pam-1.3.0-6.50.1.x86_64` ). So I presume that we will see the same issues there and consequently, the logic of skipping will make no sense anymore.

**Comment by punkioudi** [comment]
> Looks almost identical to #13307 :)

Oh indeed! Hm, what I see in that in sle, the tests don't fail yet, as  the pam version is older than the one that these 2 modules are being deprecated (eg for sle15sp3, I see `pam-1.3.0-6.50.1.x86_64` ). So I presume that we will see the same issues there and consequently, the logic of skipping will make no sense anymore.

**Comment by a-kpappas** [comment]
It looks good to me but I think someone from security should also take a look before it is merged. @lilyeyes could you also take a look? 

I also think that qe-security should be added to the maintainer field.

**Comment by lilyeyes** [comment]
Thanks for @ us and good to know this test case. Security QE do not use this "pam-test" for testing atm.
The changes LGTM. 
Btw, @rfan1 from our team takes responsibility for pam testing he would like to take a look too.

**Review by lilyeyes — APPROVED** [review]
LGTM.

File: data/pam_test/pam_deprecated.sh
Comment (rfan1) [inline data/pam_test/pam_deprecated.sh:61]: might be a typo, should be less than?

**Comment by punkioudi** [comment]
Hi @rfan1 , thanks for taking the time to review it. The purpose though of this PR is to remove two testcases of the two pam modules that have become deprecated in the newer versions of pam. So, tbh I cannot answer to your comments, as I haven't taken part in writing this testsuite :)

What I would suggest though, is that ,for now, to give priority on fixing the failure happening in TW and at the same time, the `qe-security` squad can decide on the refactoring of this test in a separate ticket.

**Comment by rfan1** [comment]
Thanks @punkioudi, that makes sense!

Can you please help check my 1st comment? it might be a typo. then I can give approve. :)

**Comment by punkioudi** [comment]
> Thanks @punkioudi, that makes sense!
> 
> Can you please help check my 1st comment? it might be a typo. then I can give approve. :)

Done :)

**Review by rfan1 — APPROVED** [review]
LGTM

**Pull request merged**
